src/charm.py

```diff
@@ -41,6 +41,7 @@ class SdcoreUpfCharm(ops.CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_status)
```

Comment:
Does this mean this fixes #12 ?

If yes add a 

"fixes #12" in the PR description, this will automatically close the issue once the pr is merged.

Reply:
No, I don't think its related (at least, it is not done to fix that bug).

Reply:
I am investigating https://github.com/canonical/sdcore-upf-operator/issues/12.

Reply:
Starting observation of collect status in the charm constructor as a first operation is a best practice. In case the remaining lines may cause a status change before collect status is not observed yet, charm could stuck in a status.